### ISSUE TITLE
NicheNet with OmnipathR on whole blood microarray data?

### ISSUE THREAD

**Issue by CosimoCristella** [saezlab/OmnipathR#26]
Hi,
first of all congratulations for the new release of such a great database and package you have built!
I'm intrigued by the new function that allows to build a nichenet model and already ran the pipeline presented in the vignette on my whole blood microarray data (bulk data). However, according to nichenetr developers, the approach could be used on "cell population – sorted bulk data", but no mention on whole blood data.

I was curious to know your opinion on this, and whether you rather recommend  using other approache (e.g. CARNIVAL)

**Comment by deeenes** [comment]
Hi,

Thanks I am glad you like it! :)

I think NicheNet makes sense only for single cell or sorted bulk data. Without distinguishing the cell types, it doesn't make sense to talk about ligand activities, except if you think that makes sense biologically, e.g. one ligand has a huge contrast in effect on all cell types, or your sample consists predominantly of one cell type. Btw, about the NicheNet pipeline in OmnipathR any feedback is very welcome. Until which stage have you run it? It's very new code, I still need to do a number of improvements.

CARNIVAL tries to find the smallest subnetwork which can explain the best a measured protein activity pattern, in your case it could be TF and pathway activities inferred by DoRothEA and PROGENy from your transcriptomics data. I think it also makes much more sense if the data is from one cell type only, but certainly there might be mechanisms which are common in the dominant cell types and these methods are able to capture from bulk data.

From bulk transcriptomics you can recover cell type specific expression by deconvolution.

I will post this to our group so maybe colleagues with more experience on these methods can comment here.

Best,

Denes

**Comment by CosimoCristella** [comment]
Hi @deeenes ,
thanks for the comments. That's exactly what I was afraid of, but curios to hear your colleagues' opinion too.
About the NicheNet pipeline, I followed instructions available on https://workflows.omnipathdb.org/nichenet1.html for autocrine signalling and managed to complete the whole procedure. Didn't try the wrapper yet, but I could test in near future.

Best,
C

**Comment by deeenes** [comment]
Hi Cosimo,

With version 2.99.19 (current tip of the master branch) I updated the NicheNet pipeline. It means I tested the whole pipeline and fixed many bugs, so from now on, it's supposed to work. You can see the vignette: https://workflows.omnipathdb.org/nichenet2.pdf and the manual: https://static.omnipathdb.org/omnipathr_manual.pdf for more info.

Finally I didn't get more ideas from colleagues, only one of them confirmed that I wrote here.

Best,

Denes

**Comment by CosimoCristella** [comment]
Hi Denes,

thanks for the update. Really nice job with the vignette! I tested the pipeline with nichenet_test() and it worked after the second try (using R version 4.0.5 on Ubuntu 20.04 machine).
Thank you also for asking around about the original question. In the meantime, I ran blind source separation analysis on my bulk gene expression data and obtained clusters of (possibly) cell-type specific markers. I'm still planning to apply your Omnipath-NicheNet pipeline on these data, although I'm still trying to figure out how to properly define transmitter and receiver populations. 

Anyway, I think we could close this issue and considered it solved, unless you have additional suggestions/remarks.

Best,

Cosimo
